sequencer/src/main/java/net/consensys/linea/rpc/services/LineaLimitedBundlePool.java

```diff
@@ -303,19 +329,33 @@ public static Hash UUIDToHash(UUID uuid) {
    */
   @Override
   public void onBlockAdded(final AddedBlockContext addedBlockContext) {
-    final var lastSeen = addedBlockContext.getBlockHeader().getNumber();
-    final var latest = maxBlockHeight.updateAndGet(current -> Math.max(current, lastSeen));
-    // keep it simple regarding reorgs and, cull the pool for any block numbers lower than latest
-    blockIndex.keySet().stream()
-        .filter(k -> k < latest)
-        // collecting to a set in order to not mutate the collection we are streaming:
-        .collect(Collectors.toSet())
-        .forEach(
-            k -> {
-              blockIndex.get(k).forEach(bundle -> cache.invalidate(bundle.bundleIdentifier()));
-              // dropping from the cache should inherently remove from blockIndex, but this
-              // is cheap insurance against blockIndex map leaking due to cache evictions
-              blockIndex.remove(k);
-            });
+    synchronized (isFrozen) {
+      if (!isFrozen.get()) { // do nothing if frozen
+        final var lastSeen = addedBlockContext.getBlockHeader().getNumber();
+        final var latest = maxBlockHeight.updateAndGet(current -> Math.max(current, lastSeen));
```

Comment:
do we need to maintain maxBlockHeight if the bundlepool now has access to chain head via blockchainService

Reply:
good catch, replaced